controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -193,6 +193,14 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 		nb.Spec.DisableLoadBalancerService = sc.Spec.MultiCloudGateway.DisableLoadBalancerService
 	}
 
+	if sc.Spec.MultiCloudGateway != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName != "" {
```

Comment:
need to check `PGSecretNamespace!=""` as well.

Reply:
done